monai/apps/mmars/mmars.py

```diff
@@ -126,27 +138,65 @@ def load_from_mmar(item, mmar_dir=None, progress: bool = True, map_location=None
     # loading with `torch.load`
     model_dict = torch.load(model_file, map_location=map_location)
     if weights_only:
-        return model_dict["model"]
-
-    # TODO: search for the module based on model name?
-    if not model_dict.get("train_conf", ""):
-        raise ValueError("The MMAR configuration does not have a 'train_conf' section.")
-    model_config = model_dict["train_conf"]["train"]["model"]
-    if model_config.get("name", ""):  # model config section is a "name"
+        return model_dict.get("model", model_dict)  # model_dict["model"] or model_dict directly
+
+    # 1. search `model_dict['train_config]` for model config spec.
+    model_config = _get_val(dict(model_dict).get("train_conf", {}), key=model_key, default={})
+    if not model_config:
+        # 2. search json CONFIG_FILE for model config spec.
+        json_path = os.path.join(model_dir, item.get(Keys.CONFIG_FILE, "config_train.json"))
+        with open(json_path) as f:
+            conf_dict = json.load(f)
+        conf_dict = dict(conf_dict)
+        model_config = _get_val(conf_dict, key=model_key, default={})
+    if not model_config:
+        # 3. search `model_dict` for model config spec.
+        model_config = _get_val(dict(model_dict), key=model_key, default={})
+
+    if not (model_config and isinstance(model_config, Mapping)):
+        raise ValueError(
+            f"Could not load model config dictionary from config: {item.get(Keys.CONFIG_FILE)}, "
+            f"or from model file: {item.get(Keys.MODEL_FILE)}."
+        )
+
+    # parse `model_config` for model class and model parameters
+    if model_config.get("name"):  # model config section is a "name"
         model_name = model_config["name"]
         model_cls = monai_nets.__dict__[model_name]
-    else:  # model config section is a "path"
+    elif model_config.get("path"):  # model config section is a "path"
         # https://docs.nvidia.com/clara/clara-train-sdk/pt/byom.html
-        model_module, model_name = model_config.get("path", "").rsplit(".", 1)
+        model_module, model_name = model_config.get("path", ".").rsplit(".", 1)
```

Comment:
Could you please help add a warning or some other notification that the model specified by "path" may be a BYOC component in the custom of MMAR, probably can't find it by `optional_import`.

Thanks.

Reply:
if the MMAR folder is on the `PYTHONPATH`, we'll be able to load it, or did I miss something here?

Reply:
It supposes that the `custom` folder is in your PYTHONPATH, when you try to run the MMAR in Clara docker, it will automatically add `custom` to PYTHONPATH.
see the `set_env.sh` in:
https://ngc.nvidia.com/catalog/models/nvidia:med:clara_pt_deepgrow_3d_annotation/files
So I think MONAI users just download the MMAR locally(don't run the MMAR) and try to build a model instance, they don't have `custom` in the PYTHONPATH.

Thanks.

Reply:
sure, let me update the error message